src/hotspot/share/cds/filemap.cpp

```diff
@@ -1914,6 +1920,42 @@ bool FileMapHeader::validate() {
                   CompactStrings   ? "enabled" : "disabled");
     return false;
   }
+  if (TrainingData::have_data()) {
+    if (_type_profile_level != TypeProfileLevel) {
+      log_info(cds)("The %s's TypeProfileLevel setting (%d)"
+                    " does not equal the current TypeProfileLevel setting (%d).", file_type,
+                    _type_profile_level, TypeProfileLevel);
+      return false;
+    }
+    if (_type_profile_width != TypeProfileWidth) {
+      log_info(cds)("The %s's TypeProfileWidth setting (%d)"
+                    " does not equal the current TypeProfileWidth setting (%d).", file_type,
+                    (int)_type_profile_width, (int)TypeProfileWidth);
+      return false;
+
+    }
+    if (_bci_profile_width != BciProfileWidth) {
+      log_info(cds)("The %s's BciProfileWidth setting (%d)"
+                    " does not equal the current BciProfileWidth setting (%d).", file_type,
+                    (int)_bci_profile_width, (int)BciProfileWidth);
+      return false;
+    }
+    if (_profile_traps != ProfileTraps) {
+      log_info(cds)("The %s's ProfileTraps setting (%s)"
+                    " does not equal the current ProfileTraps setting (%s).", file_type,
+                    _profile_traps ? "enabled" : "disabled",
+                    ProfileTraps   ? "enabled" : "disabled");
+
+      return false;
+    }
+    if (_spec_trap_limit_extra_entries != SpecTrapLimitExtraEntries) {
+      log_info(cds)("The %s's SpecTrapLimitExtraEntries setting (%d)"
+                    " does not equal the current SpecTrapLimitExtraEntries setting (%d).", file_type,
+                    _spec_trap_limit_extra_entries, SpecTrapLimitExtraEntries);
+      return false;
```

Comment:
The `log_info(cds)` should be replaced with `MetaspaceShared::report_loading_error`.  (The few `log_info` lines above this block will be fixed in [JDK-8356807](https://bugs.openjdk.org/browse/JDK-8356807))

Also, could you add a new jtreg test case for this? You can see examples in `negativeTests` in the existing AOTFlags.java test case. I think you can add your checks into the new AOTProfileFlags.java test.

Reply:
Do you want me to leave the existing `log_info` alone? Or should I fix everything in `FileMapHeader::validate()` ?

Reply:
You can leave the existing code and just fix the new code you added.

Reply:
Done. And changed the test. Please take a look.